src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/posthearingreview/PostHearingReviewSubmittedHandler.java

```diff
@@ -71,15 +71,31 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
 
         caseData = ccdCallbackMapService.handleCcdCallbackMap(callbackMap, caseData);
 
-        if (isRefusedSor(postHearing.getSetAside())) {
-            ccdService.updateCase(caseData, caseId,
-                EventType.SOR_REQUEST.getCcdType(), "Send to hearing Judge for statement of reasons", "",
-                idamService.getIdamTokens());
-        }
+        handlePostHearingRefused(caseData);
 
         return new PreSubmitCallbackResponse<>(caseData);
     }
 
+    private void handlePostHearingRefused(SscsCaseData caseData) {
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
+        PostHearing postHearing = caseData.getPostHearing();
+        if (isRefused(postHearing)) {
+            if (isYes(postHearing.getSetAside().getRequestStatementOfReasons())) {
+                ccdService.updateCase(caseData, caseId,
+                    EventType.SOR_REQUEST.getCcdType(), "Send to hearing Judge for statement of reasons", "",
+                    idamService.getIdamTokens());
+            } else {
+                ccdService.updateCase(caseData, caseId,
+                    EventType.DORMANT.getCcdType(), "Send to dormant", "",
+                    idamService.getIdamTokens());
+            }
+        }
+    }
+
+    private static boolean isRefused(PostHearing postHearing) {
```

Comment:
change this to just give the action instead of the post hearing

Reply:
The intention is to allow all post hearing types to be run through this check. On discussion, it is better for this logic to be put in the PostHearing class in common, so that it is more easily accessible. Will make that addition and replace this method.